sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/recursive/IncrementalRecursiveTests.java

```diff
@@ -0,0 +1,215 @@
+package org.dbsp.sqlCompiler.compiler.sql.recursive;
+
+import org.dbsp.sqlCompiler.circuit.operator.DBSPNestedOperator;
+import org.dbsp.sqlCompiler.compiler.CompilerOptions;
+import org.dbsp.sqlCompiler.compiler.DBSPCompiler;
+import org.dbsp.sqlCompiler.compiler.StderrErrorReporter;
+import org.dbsp.sqlCompiler.compiler.sql.tools.BaseSQLTests;
+import org.dbsp.sqlCompiler.compiler.visitors.outer.CircuitVisitor;
+import org.junit.Assert;
+import org.junit.Test;
+
+/** Tests with recursive queries and incremental compilation */
+public class IncrementalRecursiveTests extends BaseSQLTests {
+    @Override
+    public DBSPCompiler testCompiler() {
+        CompilerOptions options = this.testOptions(true, true);
+        return new DBSPCompiler(options);
+    }
+
+    @Test
+    public void testRecursive() {
+        String sql = """
+                CREATE RECURSIVE VIEW V(v INT);
+                CREATE VIEW V AS SELECT v FROM V UNION SELECT 1;""";
+        var ccs = this.getCCS(sql);
+        ccs.step("", """
+                 v | weight
+                ------------
+                 1 | 1""");
+        this.addRustTestCase(ccs);
+        CircuitVisitor visitor = new CircuitVisitor(new StderrErrorReporter()) {
+            int recursive = 0;
+            @Override
+            public void postorder(DBSPNestedOperator operator) {
+                this.recursive++;
+            }
+            @Override
+            public void endVisit() {
+                Assert.assertEquals(1, this.recursive);
+            }
+        };
+        visitor.apply(ccs.circuit);
+    }
+
+    @Test
+    public void testRecursive2() {
+        String sql = """
+                CREATE RECURSIVE VIEW V(v INT);
+                CREATE TABLE T(v INT);
+                CREATE VIEW V AS SELECT v FROM V UNION SELECT * FROM T;""";
+        var ccs = this.getCCS(sql);
+        ccs.step("", """
+                 v | weight
+                ------------""");
+        ccs.step("INSERT INTO T VALUES(1)", """
+                 v | weight
+                ------------
+                 1 | 1""");
+        ccs.step("INSERT INTO T VALUES(2)", """
+                 v | weight
+                ------------
+                 2 | 1""");
```

Comment:
Maybe I don't know how to read this, but I don't see any checks that output is correct.